### DISCUSSION TITLE
Replacing UDP transport with customized transport mechanism

### DISCUSSION THREAD

**Discussion by kunnalparihar** [eProsima/Fast-DDS#4139] [Support]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

 Hi, thanks for the great work. I am currently trying to Replace UDP transport with a customized transport mechanism (Not UDP, Not SHM and Not TCP) to communicate between two machines using the FAST DDS application layer.
I expect that data is successfully received by the subscriber's application layer.

### Current behavior

 So Far, I can send the publisher's application layer to the publisher's transport layer and then send this data subscriber's transport layer (on different machines) but unfortunately, I am not sure why I can't send data from the subscriber's transport layer to subscriber's application layer. The on_data_available function is not triggered.

### Steps to reproduce

I need your guidance on this. What could be the reason for this issue and how can i fix this.

### Fast DDS version/commit

I am using V2.12.1

### Platform/Architecture

Ubuntu Focal 20.04 arm64

### Transport layer

UDPv4

### Additional context

**I am not using UDPv4**

**Comment by JesusPoderoso** [comment]
Hi @kunnalparihar, thanks for using Fast DDS.
I am afraid that it is hard to determine the root cause of the issue with the information that you provided.
Based on your description, a tip could be to pay special atention to the TransportInterface's [OpenInputChannel](https://fast-dds.docs.eprosima.com/en/latest/fastdds/api_reference/transport/transport_generic_interfaces/transport_interface.html#_CPPv4N8eprosima7fastdds4rtps18TransportInterface16OpenInputChannelERK7LocatorP26TransportReceiverInterface8uint32_t) method, which is in charge of setting that bridge between publication's transport layer and application layer.
Check that [TransportReceiverInterface](https://fast-dds.docs.eprosima.com/en/latest/fastdds/api_reference/transport/transport_generic_interfaces/transport_receiver_interface.html#_CPPv4N8eprosima7fastdds4rtps26TransportReceiverInterfaceE) callback is properly set and called.

If you are able to provide us more information about your implementation, we may help to locate and address the issue.

**Comment by JesusPoderoso** [comment]
Also, I am moving this question to the support discussion forum as per [Fast DDS CONTRIBUTING guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md).
